regions/shapes/rectangle.py

```diff
@@ -197,6 +198,33 @@ def as_patch(self, origin=(0, 0), **kwargs):
         return Rectangle(xy=xy, width=width, height=height,
                          angle=angle, **mpl_params)
 
+    @property
+    def corners(self):
+        """
+        Return the x, y coordinate pairs that define the corners
+        """
+
+        corners = [(-self.width/2, -self.height/2),
+                   ( self.width/2, -self.height/2),
+                   ( self.width/2,  self.height/2),
+                   (-self.width/2,  self.height/2),
+                  ]
+        rotmat = [[np.cos(self.angle), np.sin(self.angle)],
+                  [-np.sin(self.angle), np.cos(self.angle)]]
+
+        return np.dot(corners, rotmat) + np.array([self.center.x,
+                                                   self.center.y])
+
+    def to_polygon(self):
+        """
+        Return a 4-cornered polygon equivalent to this rectangle
+        """
+        vertices = PixCoord(x=[c[0] for c in self.corners],
```

Comment:
You are accessing self.corners and thus computing that twice. Does this work?
```
x, y = self.corners
```
Or maybe some other Numpy array expression? But computing self.corners twice and doing Python list comprehensions seems sub-optimal.

Reply:
Completely agree that it is suboptimal.  I'll do it a better way...